manifests/init.pp

```diff
@@ -423,7 +423,7 @@
   }
 
   # The whole mysql configuration directory can be recursively overriden
-  if $mysql::source_dir {
+  if $mysql::source_dir != '' {
```

Comment:
I would change these conditionals to :
if $mysql::source_dir and if $mysql::source_dir != '' {

so that nothing is done if the value is false (boolean)